test/src/test/java/hudson/PluginManagerInstalledGUITest.java

```diff
@@ -0,0 +1,227 @@
+ */
+public class PluginManagerInstalledGUITest {
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+    
+    @Test
+    @WithPlugin("tasks.jpi")
+    public void test_enable_disable_uninstall() throws IOException, SAXException {
+        InstalledPlugins installedPlugins = new InstalledPlugins();
+        
+        InstalledPlugin tasksPlugin = installedPlugins.get("tasks");
+        InstalledPlugin cvsPlugin = installedPlugins.get("cvs");
+
+        tasksPlugin.assertHasNoDependants();
+        cvsPlugin.assertHasDependants();
+        
+        // Tasks plugin should be enabled and it should be possible to disable it
+        // because no other plugins depend on it.
+        tasksPlugin.assertEnabled();
+        tasksPlugin.assertEnabledStateChangeable();
+        tasksPlugin.assertUninstallable();
+        
+        // CVS plugin should be enabled, but it should not be possible to disable or uninstall it
+        // because the tasks plugin depends on it.
+        cvsPlugin.assertEnabled();
+        cvsPlugin.assertEnabledStateNotChangeable();
+        cvsPlugin.assertNotUninstallable();
+        
+        // Disable the tasks plugin
+        tasksPlugin.clickEnabledWidget();
+                
+        // Now the tasks plugin should be disabled, but it should be possible to re-enable it 
+        // and it should still be uninstallable.
+        tasksPlugin.assertNotEnabled(); // this is different to earlier
+        tasksPlugin.assertEnabledStateChangeable();
+        tasksPlugin.assertUninstallable();
+                
+        // The CVS plugin should still be enabled, but it should now be possible to disable it because
+        // the tasks plugin is no longer enabled. Should still not be possible to uninstall it.
+        cvsPlugin.assertEnabled();
+        cvsPlugin.assertEnabledStateChangeable(); // this is different to earlier
+        cvsPlugin.assertNotUninstallable();
+        
+        // Disable the cvs plugin
+        cvsPlugin.clickEnabledWidget();
+        
+        // Now it should NOT be possible to change the enable state of the tasks plugin because one
+        // of the plugins it depends on (the CVS plugin) is not enabled.
+        tasksPlugin.assertNotEnabled();
+        tasksPlugin.assertEnabledStateNotChangeable();  // this is different to earlier
+        tasksPlugin.assertUninstallable();
+    }
+    
+    private class InstalledPlugins {
+        
+        private final List<InstalledPlugin> installedPlugins;
+
+        private InstalledPlugins () throws IOException, SAXException {
+            JenkinsRule.WebClient webClient = jenkinsRule.createWebClient();
+            HtmlPage installedPage = webClient.goTo("pluginManager/installed");
+//            System.out.println(installedPage);
+//            try {
+//                Thread.sleep(10000000000L);
+//            } catch (InterruptedException e) { e.printStackTrace(); }
```

Comment:
I'll remove it. I was actually going to leave it coz it's handy for debugging this UI stuff - use the URL load the Jenkins in your browser to see the actual page. Of course it can be re-added.